### PR TITLE
refactor processor after NSTemplateTier changes

### PR DESCRIPTION
**Pull request by xcoulon**
No need to decode a `[]byte` into a `templatev1.Template` object in the `Process` func since the `GetNSTemplateTier` func will now return a `templatev1.Template` object.

Also, refactor some test templates to remove the unnecessary `PROJECT_NAME` parameter (won't exist in production templates)

See CRT-273

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,43 +1,81 @@
 module github.com/codeready-toolchain/member-operator
 
 require (
```
Comment (MatousJobanek) [inline go.mod]: run `go mod tidy` please

File: pkg/template/processor_test.go [hunk 2]
```diff
@@ -497,6 +482,20 @@ func assertRoleBindingExists(t *testing.T, c client.Client, ns string) authv1.Ro
 	return rb
 }
 
+type decodeFunc func([]byte, *schema.GroupVersionKind, runtime.Object) (runtime.Object, *schema.GroupVersionKind, error)
```
Comment (MatousJobanek) [inline pkg/template/processor_test.go]: this is not used anywhere

File: go.mod (hunk 1 above)
Comment (xcoulon) [reply go.mod]: done in e165b0b

File: pkg/template/processor_test.go (hunk 2 above)
Comment (xcoulon) [reply pkg/template/processor_test.go]: ah right, that's a left-over :/ I removed it in e165b0b

**Comment by xcoulon** [comment]
thanks for the review @alexeykazakov 🙌

**Comment by xcoulon** [comment]
thanks for the review @nurali-techie !

**Pull request merged**